service/policies.go

```diff
@@ -0,0 +1,86 @@
+package service
+
+import (
+	"sync"
+
+	v1 "code.vegaprotocol.io/protos/vega/wallet/v1"
+	"code.vegaprotocol.io/vegawallet/crypto"
+	"github.com/golang/protobuf/jsonpb"
+)
+
+type ConsentConfirmation struct {
+	TxStr    string
+	Decision bool
+}
+
+type ConsentRequest struct {
+	tx            *v1.SubmitTransactionRequest
+	Confirmations chan ConsentConfirmation
+}
+
+func (r *ConsentRequest) String() (string, error) {
+	m := jsonpb.Marshaler{Indent: "    "}
+	marshalledRequest, err := m.MarshalToString(r.tx)
+	return marshalledRequest, err
+}
+
+type Policy interface {
+	Ask(tx *v1.SubmitTransactionRequest) bool
+	GetSignRequestsConfirmations(hash string) chan ConsentConfirmation
+}
+
+type AutomaticConsentPolicy struct {
+	pendingEvents chan ConsentRequest
+}
+
+func NewAutomaticConsentPolicy(pending chan ConsentRequest) Policy {
+	return &AutomaticConsentPolicy{
+		pendingEvents: pending,
+	}
+}
+
+func (p *AutomaticConsentPolicy) Ask(tx *v1.SubmitTransactionRequest) bool {
+	return true
+}
+
+func (p *AutomaticConsentPolicy) GetSignRequestsConfirmations(hash string) chan ConsentConfirmation {
+	return nil
+}
+
+type ExplicitConsentPolicy struct {
+	pendingEvents    chan ConsentRequest
+	AllConfirmations sync.Map
+}
+
+func NewExplicitConsentPolicy(pending chan ConsentRequest) Policy {
+	return &ExplicitConsentPolicy{
+		pendingEvents:    pending,
+		AllConfirmations: sync.Map{},
+	}
+}
+
+func (p *ExplicitConsentPolicy) GetSignRequestsConfirmations(hash string) chan ConsentConfirmation {
+	confirmations, ok := p.AllConfirmations.Load(hash)
+	if ok {
+		return confirmations.(chan ConsentConfirmation)
+	}
+	return nil
+}
+
+func (p *ExplicitConsentPolicy) Ask(tx *v1.SubmitTransactionRequest) bool {
+	txHash := crypto.AsSha256(tx)
+	confirmations := make(chan ConsentConfirmation)
+	p.AllConfirmations.Store(txHash, confirmations)
```

Comment:
From what I can see `p.AllConfirmations` is just used for unittest. If so, we don't need it. Everything can be tested without it by doing two things:
1) create a mock of the `Policy` interface and pass it into the service in `getTestService`. Then when your test calls `Ask` you response approriately. This will flex the code in `signTx`.
2) Make a new test that instantiates just a `ExplicitConsetPolicy` where you pass in the pending-channel. Then your test can read from that channel in a go-routine ready with a response, and then you can call `Ask`. This will flex the code here.

Reply:
OK, I thought that confirmations could be used in the future to monitor at the console level or for debug in live scenarios